wrangler-core/src/main/java/io/cdap/wrangler/metrics/DirectiveWithJEXLCategoryMetric.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.metrics;
+
+import com.google.common.collect.ImmutableMap;
+import io.cdap.wrangler.api.EntityWithMetrics;
+import io.cdap.wrangler.api.WranglerMetric;
+import io.cdap.wrangler.expression.EL;
+
+import java.util.Map;
+
+import static io.cdap.cdap.api.metrics.MetricType.COUNTER;
+import static io.cdap.cdap.common.conf.Constants.Metrics.Tag.APP_ENTITY_TYPE;
+import static io.cdap.cdap.common.conf.Constants.Metrics.Tag.APP_ENTITY_TYPE_NAME;
+import static io.cdap.cdap.common.conf.Constants.Metrics.Tag.SERVICE;
+
+/**
+ * Represents a directive that emits the 'JEXL category' metric. This metric counts the number of times
+ * a JEXL function category in {@link io.cdap.wrangler.expression.EL.DefaultFunctions} is used.
+ */
+public interface DirectiveWithJEXLCategoryMetric extends EntityWithMetrics {
```

Comment:
nit: DirectiveWithJEXLCategoryMetric can be DirectiveJEXLCategoryMetric or JEXLCategoryDirectiveMetric